jina/serve/runtimes/worker/__init__.py

```diff
@@ -156,6 +152,24 @@ async def _async_setup_grpc_server(self):
 
     async def async_run_forever(self):
         """Block until the GRPC server is terminated"""
+        if self.args.hot_reload:
+            import inspect
+
+            watched_files = [
+                inspect.getfile(self._request_handler._executor.__class__)
+            ] + (self.args.py_modules or [])
+            with ImportExtensions(
+                required=True,
+                logger=self.logger,
+                help_text='''hot reload requires jina in development mode, you can do: `pip install jina[devel]''',
+            ):
+                from watchfiles import awatch
+
+            async for changes in awatch(*watched_files):
```

Comment:
u should creatr another task that runs asynchronously, but this way the server will not run